plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -354,10 +356,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         List<SchemaTableName> tablesList = schemaName.map(Collections::singletonList)
                 .orElseGet(metastore::getAllDatabases)
                 .stream()
-                .flatMap(schema -> metastore.getTablesWithParameter(schema, TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE).stream()
-                        .map(table -> new SchemaTableName(schema, table))
-                        .collect(toList())
-                        .stream())
+                .flatMap(schema -> Stream.concat(
+                        metastore.getTablesWithParameter(schema, TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE.toLowerCase(Locale.ENGLISH)).stream()
```

Comment:
can we avoid using two HMS calls here using a disjunctive filter? look at documentation in https://github.com/apache/hive/blob/68a759a9925efe5f04d6dfe8f4a9ccc3d7a068e6/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java#L380

[This PR](https://github.com/trinodb/trino/pull/6595) tries to do it, but it's still a WIP so not 100% sure if it works as expected: https://github.com/trinodb/trino/pull/6595/commits/5c98076f5f1cbccac31f92daf19ef3c7ca934da9#diff-83c934e2538dd1e52ea1dab72378428812209855665ae182bdf523d655ad218eR949

Reply:
This usecase seems to be different from the linked PR in the sense that the change here needs to be at a higher level (in `HiveMetastore` interface). The filter can involve any tree of AND and OR operartors, and so the api design change here will likely need more discussion than just making a small change in `ThriftHiveMetastore`. I'm tending towards prioritizing the correctness (over the performance impact of two calls) for now.